### PR TITLE
iPhone X support and additional Safe Area insets support

### PR DESCRIPTION
**Pull request by toohotz**

### CODE REVIEW

**Comment by himaratsu** [comment]
@toohotz Thank your for Gread PR!
I will merge it if https://github.com/PureLayout/PureLayout/pull/203 discussion is closed.

**Comment by toohotz** [comment]
No problem 👍🏾, you might want the changes for yourself as well as it fixes the things with the trailing and bottom to superview along with the inequalities and other methods as well.  The maintainer of the repo seems like he is going to reach out for some others to help keep things _alive_ but I'll drop a note there referring to this PR as well.

**Comment by toohotz** [comment]
Hey there, are you going to integrate these changes into your fork?

**Comment by himaratsu** [comment]
@toohotz
Hey, I'm back. I will merge your PR in day or two. Thanks!

File: PureLayout/PureLayout/ALView+PureLayout.m [hunk 1]
```diff
@@ -184,24 +197,88 @@ - (NSLayoutConstraint *)autoPinEdgeToSuperviewSafeArea:(ALEdge)edge withInset:(C
...
+            }
             break;
         case ALEdgeRight:
-            constraint = [[self rightAnchor] constraintEqualToAnchor:rightAnchor constant:inset];
+            switch (relation) {
+                case NSLayoutRelationEqual:
+                    constraint = [[self rightAnchor] constraintEqualToAnchor:rightAnchor constant:inset];
+                    break;
+                case NSLayoutRelationLessThanOrEqual:
+                    constraint = [[self leftAnchor] constraintGreaterThanOrEqualToAnchor:rightAnchor constant:inset];
```
Comment (himaratsu) [inline PureLayout/PureLayout/ALView+PureLayout.m]: @toohotz Is this mistake of `rightAnchor` ?

I think it expect:
```swift
constraint = [[self rightAnchor] constraintGreaterThanOrEqualToAnchor:rightAnchor constant:inset];
```

If not, Please tell me your intension.

Comment (toohotz) [reply PureLayout/PureLayout/ALView+PureLayout.m]: Good catch, copy pasted had caused the issue, will fix it thank you for catching.

Comment (himaratsu) [reply PureLayout/PureLayout/ALView+PureLayout.m]: Thanks. Other code feel good, so I will merge as soon as it's fixed.

Comment (toohotz) [reply PureLayout/PureLayout/ALView+PureLayout.m]: No problem, done.

**Comment by himaratsu** [comment]
Thanks!

**Comment by jjksam** [comment]
Hey guys, I found your approach using NSLayoutAnchor is great, some how I found that there is some compatibility problem if targeting iOS 8.0, the NSLayoutAnchor is only available in iOS 9.0, so if I run the app in iOS 8 (pretty old system), it will cause crash. Although PureLayout do claim that fully compatible with iOS 9.0 but at least it won't crash even if it target the iOS 8.0.
Should we use the primitive API instead of higher level API until the project should no longer support such low version system ? Or whenever higher API available, we can make use of the higher level API.

**Comment by toohotz** [comment]
@jjksam Agreed, `NSLayoutAnchor` was introduced in iOS 9 so we should restrict this API to target iOS 9 and forward.  Since the iPhone X is the first device that utilizes safe area insets and it started life on iOS 11, it should pose no issues.  This will be done on the PR on the PureLayout project.

**Pull request merged**